template-for-repository/proofs/run-cbmc-proofs.py

```diff
@@ -0,0 +1,268 @@
+    else:
+        level = logging.WARNING
+    logging.basicConfig(
+        format="run-cbmc-proofs: %(message)s", level=level)
+
+
+def task_pool_size():
+    ret = os.cpu_count()
+    if ret is None or ret < 3:
+        return 1
+    return ret - 2
+
+
+def print_counter(counter):
+    print(
+        "\rConfiguring CBMC proofs: "
+        "{complete:{width}} / {total:{width}}".format(
+            **counter), end="", file=sys.stderr)
+
+
+def get_proof_dirs(proof_root, proof_list, proof_marker):
+    if proof_list is not None:
+        proofs_remaining = list(proof_list)
+    else:
+        proofs_remaining = []
+
+    for root, _, fyles in os.walk(proof_root):
+        proof_name = str(pathlib.Path(root).name)
+        if proof_list and proof_name not in proof_list:
+            continue
+        if proof_list and proof_name in proofs_remaining:
+            proofs_remaining.remove(proof_name)
+        if proof_marker in fyles:
+            yield root
+
+    if proofs_remaining:
+        logging.error(
+            "The following proofs were not found: %s",
+            ", ".join(proofs_remaining))
+        sys.exit(1)
+
+
+def run_build(litani, jobs):
+    cmd = [str(litani), "run-build"]
+    if jobs:
+        cmd.extend(["-j", str(jobs)])
+
+    logging.debug(" ".join(cmd))
+    proc = subprocess.run(cmd)
+    if proc.returncode:
+        logging.error("Failed to run litani run-build")
+        sys.exit(1)
+
+
+def get_litani_path(proof_root):
+    cmd = [
+        "make",
+        "PROOF_ROOT=%s" % proof_root,
+        "-f", "Makefile.common",
+        "litani-path",
+    ]
+    logging.debug(" ".join(cmd))
+    proc = subprocess.run(cmd, universal_newlines=True, stdout=subprocess.PIPE)
+    if proc.returncode:
+        logging.error("Could not determine path to litani")
+        sys.exit(1)
+    return proc.stdout.strip()
+
+
+async def configure_proof_dirs(queue, counter):
+    while True:
+        print_counter(counter)
+        path = str(await queue.get())
+
+        proc = await asyncio.create_subprocess_exec(
+            "nice", "-n", "15", "make", "-B", "--quiet", "_report", cwd=path)
+        await proc.wait()
+        counter["fail" if proc.returncode else "pass"].append(path)
+        counter["complete"] += 1
+
+        print_counter(counter)
+        queue.task_done()
+
+
+async def main():
+    args = get_args()
+    set_up_logging(args.verbose)
+
+    proof_root = pathlib.Path(__file__).resolve().parent
+    litani = get_litani_path(proof_root)
+
+    if not args.no_standalone:
+        cmd = [str(litani), "init", "--project", args.project_name]
+        logging.debug(" ".join(cmd))
+        proc = subprocess.run(cmd)
+        if proc.returncode:
+            logging.error("Failed to run litani init")
+            sys.exit(1)
+
+    proof_dirs = list(get_proof_dirs(
+        proof_root, args.proofs, args.proof_marker))
+    if not proof_dirs:
+        logging.error("No proof directories found")
```

Comment:
I think we should use `logging.critical` since we cannot recover from this error and we call `sys.exit(1)` next.